test/unit/upload.js

```diff
@@ -36,66 +41,218 @@ describe('Upload Methods', function () {
       }
       mockUser = {}
       mockUser.newInstance = sinon.stub().returns(mockInstance)
-      mockArgs = { _user: mockUser, file: 'mockFile.txt' }
-      sinon.stub(utils, 'getRepositoryAndInstance').resolves([ mockArgs, mockInstance ])
+      sinon.stub(utils, 'getRepositoryAndInstance')
       sinon.stub(fs, 'readFile').yieldsAsync(null, mockFileData)
+      sinon.spy(upload, '_createFile')
+      sinon.spy(upload, '_recusivelyCreateDirectories')
     })
 
     afterEach(function () {
       utils.getRepositoryAndInstance.restore()
       fs.readFile.restore()
+      upload._createFile.restore()
+      upload._recusivelyCreateDirectories.restore()
```

Comment:
(oh, `_createFile` is called by `uploadFile` - got it. just have to make sure it's called! :smile:)